### PR TITLE
DFHack::Units: new function `setGoal`

### PR DESCRIPTION
**Pull request by chdoc**
Discussion: https://discord.com/channels/793331351645323264/873659717174587403/1287010527020585001

### CODE REVIEW

**Comment by quietust** [comment]
I'm thinking "setPathGoal" might be a better name for this, because there are multiple types of Goals in the game (most notably Personality Goals).

**Comment by chdoc** [comment]
Rebased on top of 50.14-r1.1 (and renamed as suggested)

File: docs/changelog.txt [hunk 1]
```diff
@@ -113,6 +113,7 @@ Template for new versions:
 - ``Units``: new ``isWildlife`` and ``isAgitated`` property checks
 - ``Items::createItem``: removed ``growth_print`` parameter; now determined automatically
 - ``DFHack::cuboid``: ``cuboid::clampMap`` now returns the cuboid itself (instead of boolean) to allow method chaining; call ``cuboid::isValid`` to determine success
+- ``DFHack::Units``: new function ``setPathGoal``
```
Comment (myk002) [inline docs/changelog.txt:116]: should be moved up to the "Future" section above. Also, the Lua variant should be mentioned in the Lua section

File: docs/dev/Lua API.rst [hunk 2]
```diff
@@ -1703,6 +1703,11 @@ Units module
   ``dfhack.units.isOwnCiv`` or another appropriate predicate on the unit
   in question.
 
+* ``dfhack.units.setGoal(unit, pos, goal)``
```
Comment (myk002) [inline docs/dev/Lua API.rst:1706]:
```suggestion
* ``dfhack.units.setPathGoal(unit, pos, goal)``
```

File: library/modules/Units.cpp [hunk 3]
```diff
@@ -933,6 +934,28 @@ void Units::makeown(df::unit *unit) {
...
+    {
+        unit->path.dest = pos;
+        unit->path.goal = goal;
+        unit->path.path.clear();
+    }
+
+    if (unit->flags1.bits.rider && unit->mount_type == df::rider_positions_type::STANDARD)
+    {
+        auto mount = df::unit::find(unit->relationship_ids[df::unit_relationship_type::RiderMount]);
+        if (mount)
```
Comment (myk002) [inline library/modules/Units.cpp:950]:
```suggestion
        if (auto mount = df::unit::find(unit->relationship_ids[df::unit_relationship_type::RiderMount]))
```

**Review by myk002 — APPROVED** [review]

**Pull request merged**